src/test/java/seedu/address/logic/commands/UpcomingCommandTest.java

```diff
@@ -32,44 +36,108 @@ public void constructor_nullDeliveryIsUpcomingPredicate_throwsNullPointerExcepti
         assertCommandSuccess(command, model, expectedMessage, expectedModel);
         assertEquals(Collections.emptyList(), model.getFilteredDeliveryList());
     }
 
     //need check with JZ why this fails
     @Test
-    public void execute_lateCompletionDate_multipleDeliveriesFound() {
+    public void execute_deliveryBeforeCompletionDate_multipleDeliveriesFound() {
         String expectedMessage = String.format(MESSAGE_DELIVERIES_LISTED_OVERVIEW, 3);
-        DeliveryIsUpcomingPredicate predicate = preparePredicate("31-12-2030 23:59");
-        UpcomingCommand command = new UpcomingCommand(predicate);
+        DeliveryIsUpcomingBeforePredicate predicate = prepareBeforePredicate("31-12-2030 23:59");
+        List<Predicate<Delivery>> predicates = new ArrayList<>();
+        predicates.add(predicate);
+        UpcomingCommand command = new UpcomingCommand(predicates);
         expectedModel.updateFilteredDeliveryList(predicate);
         assertCommandSuccess(command, model, expectedMessage, expectedModel);
         assertEquals(expectedModel.getFilteredDeliveryList(), model.getFilteredDeliveryList());
     }
 
+    @Test
+    public void execute_deliveryAfterStartDate_multipleDeliveriesFound() {
+        String expectedMessage = String.format(MESSAGE_DELIVERIES_LISTED_OVERVIEW, 3);
+        DeliveryIsUpcomingAfterPredicate predicate = prepareAfterPredicate("31-12-2001 23:59");
+        List<Predicate<Delivery>> predicates = new ArrayList<>();
+        predicates.add(predicate);
+        UpcomingCommand command = new UpcomingCommand(predicates);
+        expectedModel.updateFilteredDeliveryList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(expectedModel.getFilteredDeliveryList(), model.getFilteredDeliveryList());
+    }
+    @Test
+    public void execute_deliveryBeforeStartDate_noDeliveriesFound() {
+        String expectedMessage = String.format(MESSAGE_DELIVERIES_LISTED_OVERVIEW, 0);
+        DeliveryIsUpcomingAfterPredicate predicate = prepareAfterPredicate("31-12-2030 23:59");
+        List<Predicate<Delivery>> predicates = new ArrayList<>();
+        predicates.add(predicate);
+        UpcomingCommand command = new UpcomingCommand(predicates);
+        expectedModel.updateFilteredDeliveryList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(expectedModel.getFilteredDeliveryList(), model.getFilteredDeliveryList());
+    }
+
+    @Test
+    public void execute_betweenStartAndEndDate_multipleDeliveriesFound() {
+        String expectedMessage = String.format(MESSAGE_DELIVERIES_LISTED_OVERVIEW, 3);
+        DeliveryIsUpcomingBeforePredicate predicateBefore = prepareBeforePredicate("04-12-2025 23:59");
+        DeliveryIsUpcomingAfterPredicate predicateAfter = prepareAfterPredicate("12-10-2020 07:30");
+        List<Predicate<Delivery>> predicates = new ArrayList<>();
+        predicates.add(predicateAfter);
```

Comment:
Should there be a method to create a predicate with both before and after dates?